### DISCUSSION TITLE
I can't to fork my own repository

### DISCUSSION THREAD

**Discussion by VovanR** [community/community#55138] [Repositories]
### Select Topic Area

Bug

### Body

When I trying to fork my own repository it says me that I already have repository with this name. But if I set another unused repository name - nothing changed. This interface is not allowed me to fork my own repository =(
![Peek 2023-05-11 10-04](https://github.com/community/community/assets/430003/8384a0e6-cdc0-40af-9ae3-8fcea30f0b9f)

**Comment by hrishikesh-kocharekar-dev** [comment]
Hi Vladimir,

Go to your profile on the platform where your repository is located (e.g., GitHub).
Click on the "New repository" button.
Give your new repository a name and a description (if desired).
Choose the repository type (e.g., public or private).
Click "Create repository."
Once you have created your new repository, you can clone the code from your existing repository and push it to the new repository using Git. Here are the steps to do that:

Open a terminal or Git Bash window.
Navigate to the directory where your existing repository is located using the `cd` command.
Clone your existing repository using the command `git clone <repository URL>`.
Navigate into the newly created directory using `cd <repository name>`.
Initialize a new Git repository using `git init`.
Add your files using `git add .`.
Commit your changes using `git commit -m "Initial commit"`.
Add the URL of your new repository as a remote using git `remote add origin <new repository URL>`.
Push your changes to the new repository using `git push -u origin master`.
These steps will effectively create a copy of your existing repository in a new repository with a different name.

**Reply by humanplayer2** [reply]
Can this be used to create a merge request from the new repository to the old?

**Comment by Yarden-zamir** [comment]
You can import the repo using [github importer](https://github.blog/2016-02-11-migrate-your-code-with-the-github-importer/)
<img width="1328" alt="Screen Shot 2023-05-11 at 11 39 06" src="https://github.com/community/community/assets/8178413/c981c0c8-ddaf-4065-bb66-dc6cf543552a">

<img width="668" alt="Screen Shot 2023-05-11 at 11 39 44" src="https://github.com/community/community/assets/8178413/8c8a6e86-6cb7-4d0c-834d-b93674f44d79">

**Comment by ToasterUwU** [comment]
Any way to actually fork and not just copy paste the contents to a new repo? I specifically need to have "repo b" be a fork of "repo a".

Specifcially i have a template which gets regular updates from me, and i make other things from it. 
And i want to easily apply fixes and all that to my forks, while having them separate because they are their own project, just based on the original template repo. 

And using the actual "Template" function from Github doest work as far as i know, since i cant make a PR from it like i could with a fork

**Comment by bigjosh** [comment]
It seems the only way to fork your own repo is to make an organization in your account and then use that to make the fork...

https://gist.githubusercontent.com/drcode/02a9a38cc3fc2400af8b9bc3789ba0e6/raw/b9040f6e2f18dc4eafc8ff4e2e90bc3d876b399a/gistfile1.txt

**Comment by VladimirCreator** [comment]
+1
GitHub really needs it

**Comment by dannevesdantas** [comment]
A curious question: If it works for public and private organizations why it doesn't work for personal accounts? 🤔

The ability to fork your own repo is important because, if you add new commits to the base repo, the forks can sync with it.

**Comment by YeesterPlus** [comment]
create a new branch

**Comment by RobertFent** [comment]
I had the same problem. Fixed it with this workaround:
1. clone repo into own repo dir
```bash
git clone {repo to fork} {folder for your repo}
```
2. set remote to own repo and push data
```bash
git remote rename origin upstream
git remote add origin {your repo}
git push -u origin master
```
4. pull from "forked" repo
```bash
git fetch upstream
git merge upstream/master
```
Now when you push in your repo it will only add changes there.<br>
Fetching from upstream and merging is kind of the same like pulling from given upstream branch
